src/cpuinfo_x86.c

```diff
@@ -1139,10 +1143,13 @@ static void ParseLeaf2(const int max_cpuid_leaf, CacheInfo* info) {
   }
 }
 
-static void ParseLeaf4(const int max_cpuid_leaf, CacheInfo* info) {
+// For newer Intel CPUs uses "CPUID, eax=0x00000004".
+// For newer AMD CPUs uses "CPUID, eax=0x8000001D"
+static void ParseCacheInfo(const int max_cpuid_leaf, uint32_t leaf_id,
+                           CacheInfo* info) {
   info->size = 0;
   for (int cache_id = 0; cache_id < CPU_FEATURES_MAX_CACHE_LEVEL; cache_id++) {
-    const Leaf leaf = SafeCpuIdEx(max_cpuid_leaf, 4, cache_id);
+    const Leaf leaf = SafeCpuIdEx(max_cpuid_leaf, leaf_id, cache_id);
     CacheType cache_type = ExtractBitRange(leaf.eax, 4, 0);
```

Comment:
`ExtractBitRange(leaf.eax, leaf_id, 0);` ?

Reply:
@Mizux if you propose to use ExtractBitRange(leaf.eax, leaf_id, 0), no, we cannot because for AMD CPUs leaf_id is 0x8000001D, so we will not be able to get the type of cache. Or did I not understand the question?

Reply:
my bad both `4` are unrelated, sorry for the noise !